ql/src/test/org/apache/hadoop/hive/ql/io/parquet/serde/TestParquetTimestampsHive2Compatibility.java

```diff
@@ -97,6 +97,21 @@ void testWriteHive2ReadHive4UsingLegacyConversionWithZone(String timestampString
     }
   }
 
+  @ParameterizedTest(name = "{0}")
+  @MethodSource("generateJulianLeapYearTimestamps")
+  void testWriteHive2ReadHive4UsingLegacyConversionWithJulianLeapYears(String timestampString) {
+    TimeZone original = TimeZone.getDefault();
+    try {
+      String zoneId = "Asia/Singapore";
```

Comment:
please check edge cases: UTC+12 write UTC-12 read, and reversed

Reply:
Done